### PR TITLE
Add Updates

### LINKED ISSUE #6: SDK: Add Fallback apr calculation

### LINKED ISSUE #9: Add APR and update its calculation

### LINKED ISSUE #10: SDK: check bug in params

### PR DESCRIPTION
**Pull request by ghost**
### Description of changes proposed in this pull request:
- [x] Update documentation
- [x] Add gcr calculation
- [x] Add current position cr calculation
- [x] Add apr calculation
- [x] Add stats fallback

### Checks:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: src/library/Asset.ts [hunk 1]
```diff
@@ -37,15 +37,17 @@ export class Asset {
...
-  // TODO getGCR
-  // getGCR = async () => {
-  //   return await this.methods.getGCR(this.asset);
-  // };
+  getPositionCR = async () => {
+    return await this.methods.getPositionCR(this.asset);
+  };
+
+  getUserStats = async () => {
+    return await this.methods.getUserStats(0, 1621298935)
```
Comment (ethedev) [inline src/library/Asset.ts:45]: any reason this is passed here? `1621298935` can just be handled inside the function itself

File: src/library/AssetMethods.ts [hunk 2]
```diff
@@ -122,16 +122,24 @@ export class AssetMethods {
...
   * @public
   * @methods
   */
-  getUserStats = async (interval: string, startDate: string, endDate: string) => {
-    // console.debug("sdk getUserStats", interval, startDate, endDate);
-    const stats = await getUserTxStats(this.options.provider, this.options.account, interval, startDate, endDate)
-    return stats;
+  getUserStats = async (startTimestamp: number, endTimestamp: number) => {
+    startTimestamp = 0;
+    endTimestamp = 1621298935;
```
Comment (ethedev) [inline src/library/AssetMethods.ts:132]: im assuming this will overwrite the `endTimestamp` which was also not set up correctly on the Asset.ts file? this should be changed

File: src/library/AssetMethods.ts [hunk 3]
```diff
@@ -122,16 +122,24 @@ export class AssetMethods {
...
   * @param {string} startDate Start date of the input
   * @param {string} endDate End date of the input
   * @public
   * @methods
   */
-  getUserStats = async (interval: string, startDate: string, endDate: string) => {
-    // console.debug("sdk getUserStats", interval, startDate, endDate);
-    const stats = await getUserTxStats(this.options.provider, this.options.account, interval, startDate, endDate)
-    return stats;
+  getUserStats = async (startTimestamp: number, endTimestamp: number) => {
```
Comment (ethedev) [inline src/library/AssetMethods.ts:130]: because this is not related to any asset, it shouldnt be on this file, this file is only assets methods related

File: src/library/AssetMethods.ts (hunk 2 above)
Comment (ethedev) [reply src/library/AssetMethods.ts:134]: this will overwrite the `endTimestamp` which was also not set up correctly on the Asset.ts file? this should be changed

Comment (byterose) [reply src/library/AssetMethods.ts]: Related to param bug

File: tests/sdk.test.ts [hunk 4]
```diff
@@ -12,8 +13,11 @@ const account = "0x3f5ce5fbfe3e9af3971dd833d26ba9b5c936f0be";
 beforeAll(async () => {
   const web3 = await new Web3(new Web3.providers.HttpProvider(process.env.INFURA_URL_HTTP || ""));
   const provider: ExternalProvider = (new Web3Provider(web3.currentProvider as any)).provider;
+  const ethersProvider = new ethers.providers.JsonRpcProvider(process.env.INFURA_URL_HTTP || "");
+  /// @notice Passing ethersProvider is just for apr integration on v2
   SDK = await new Degenerative({
     provider: provider,
+    ethersProvider: ethersProvider,
```
Comment (ethedev) [inline tests/sdk.test.ts:20]: we dont need to have 2 different providers but only 1 provider which should work despite how it looks like, you can convert providers if that is nessesary after passing it through `provider`

File: tests/sdk.test.ts [hunk 5]
```diff
@@ -61,4 +65,39 @@ describe('General function testing', () => {
...
+  });
+
+  it("getPositionCR must return an string", async () => {
+    const pos = await SDK.methods.getPositionCR(Assets[network]["ugas"][0]);
+    // console.debug("pos", pos)
+    expect(typeof pos).toBe("string");
+  });
+
+  it("getAPR should return a number", async () => {
+    const apr = await SDK.methods.getAPR('80', '1.5');
```
Comment (ethedev) [inline tests/sdk.test.ts:81]:
```suggestion
    // const apr = await SDK.apr.getMiningRewards('uGAS-JUN21', Assets[network]["ugas"][3], 107.5);
    const apr = await SDK.methods.getAPR('80', '1.5');
```

File: tests/sdk.test.ts [hunk 6]
```diff
@@ -61,4 +65,39 @@ describe('General function testing', () => {
...
+  /**
+   * @notice The following method tests should be abandoned in the future.
+   * To see the test results, please uncomment them.
+   */
+  // it("getAPR should return a number", async () => {
+  //   /// @TODO Pass right params
+  //   const apr = await SDK.apr.getMiningRewards('uGAS-JUN21', Assets[network]["ugas"][3], 107.5);
+  //   // console.debug("apr", apr)
+  //   expect(typeof apr).toBe("string");
+  // });
```
Comment (ethedev) [inline tests/sdk.test.ts:95]:
```suggestion

```

File: src/library/TxStats.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  /**
+  * Fetch user transactions statistics
+  * @param {number} startTimestamp Start timestamp of the input
+  * @param {number} endTimestamp End timestamp of the input
+  * @public
+  * @methods
+  */
+  getUserStats = async (startTimestamp: number, endTimestamp: number) => {
+    startTimestamp = 1623619086;
+    endTimestamp = 1623885486;
```
Comment (ethedev) [inline src/library/TxStats.ts:25]: `startTimestamp` `endTimestamp` are overwritten here, if the param bug is fixed this as well shouldve been updated

**Review by ethedev — CHANGES_REQUESTED** [review]
